### ISSUE TITLE
IDE0018 when using out variables

### ISSUE THREAD

**Issue by ArgTang** [dotnet/roslyn#17107]
**Version Used**: 
Latest VS 2017, not sure how to see witch Roslyn version is used
**Steps to Reproduce**:
```C#
Dictionary<Bar, List<Foo>> dict,
List<Foo> fooList ;
if(!dict.TryGetValue(date, out fooList )) // IDE0018 here
{
    fooList = new List<Foo>();
    dict.Add(date, fooList );
}
```

**Expected Behavior**:
If value is used with an out parameter, IDE0018 should be ignored

**Actual Behavior**:
IDE0018 is raised when var has an `out`

https://developercommunity.visualstudio.com/content/problem/8924/ide0018-false-suggestion.html

**Comment by gafter** [comment]
The suggestion to inline the variable declaration looks perfectly reasonable in this context, as variables may be declared in an `out` parameter in C# 7.

**Comment by miloush** [comment]
@gafter the issue at the link is a bit different, it's a `foreach` and reuses the out variable after the cycle.

**Comment by gafter** [comment]
@miloush That may be true. I'm commenting on *this* issue.

**Comment by ArgTang** [comment]
You are right, i haven't used that feature before. The (google/bing) search for IDE0018 does not give a good answer yet.
